tasks/eva_2326/src/main/groovy/eva2326/update_contig_name.groovy

```diff
@@ -0,0 +1,596 @@
+
+        // Obtain a MongoCursor to iterate through documents
+        def mongoCursor = mongoTemplate.getCollection(VARIANTS_COLLECTION)
+                .find()
+                .noCursorTimeout(true)
+                .batchSize(BATCH_SIZE)
+                .iterator()
+
+        List<VariantDocument> variantBatch = new ArrayList<>()
+
+        while (mongoCursor.hasNext()) {
+            Document orgDocument = mongoCursor.next()
+            VariantDocument orgVariant
+            try {
+                orgVariant = mongoTemplate.getConverter().read(VariantDocument.class, orgDocument)
+            } catch (Exception e) {
+                logger.error("Exception while converting Bson document to Variant Document with _id: {} " +
+                        "chr {} start {} ref {} alt {}. Exception: {}", orgDocument.get("_id"),
+                        orgDocument.get("chr"), orgDocument.get("start"), orgDocument.get("ref"),
+                        orgDocument.get("alt"), e.getMessage())
+
+                storeVariantsThatCantBeProcessed(variantsWithIssuesFilePath, orgDocument.get("_id"), "", "Exception converting Bson Document to Variant Document")
+                orgVariant = null
+            }
+
+            if (orgVariant != null) {
+                variantBatch.add(orgVariant)
+                BATCH_COUNT++
+            }
+
+            if (BATCH_COUNT >= BATCH_SIZE) {
+                processVariantBatch(variantBatch)
+                TOTAL_COUNTS += BATCH_COUNT
+                logger.info("Total document processed till now: {}", TOTAL_COUNTS)
+                variantBatch.clear()
+                BATCH_COUNT = 0
+            }
+        }
+
+        if (variantBatch.size() > 0) {
+            processVariantBatch(variantBatch)
+            TOTAL_COUNTS += BATCH_COUNT
+            logger.info("Total document processed till now: {}", TOTAL_COUNTS)
+            variantBatch.clear()
+            BATCH_COUNT = 0
+        }
+
+        // Finished processing
+         System.exit(0)
+    }
+
+    void processVariantBatch(List<VariantDocument> variantDocumentList) {
+
+        while (!variantDocumentList.isEmpty()) {
+            Set<String> idProcessed = new HashSet<>()
+            List<VariantDocument> processInNextLoop = new ArrayList<>()
+
+
+            // create a map of original variant id and its INSDC accession chromosome
+            Map<String, String> orgVariantInsdcChrMap = variantDocumentList.stream()
+                    .collect(Collectors.toMap(orgVariant -> orgVariant.getId(), orgVariant -> {
+                        String orgChromosome = orgVariant.getChromosome()
+                        StringBuilder reason = new StringBuilder()
+                        ContigSynonyms contigSynonyms = contigMapping.getContigSynonyms(orgChromosome)
+                        if (contigMapping.isGenbankReplacementPossible(orgChromosome, contigSynonyms, reason)) {
```

Comment:
We've been using a more lenient version of this method, see [here](https://github.com/EBIvariation/eva-tasks/blob/ae261b7539a94414b30242cb188d47b7ecfc9486/tasks/eva_3660/src/main/groovy/eva3660/ContigRenamingProcessor.groovy#L60).

It actually might be worth re-using that class entirely, as it includes at least one set of workarounds (contig names in DB not present in assembly report) that we needed for contig mapping last time.

Reply:
Done